### PR TITLE
[release-0.56] sriov: Fix network to PCI-Address mapping

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This is a manual cherry-pick of the following PRs:
- https://github.com/kubevirt/kubevirt/pull/8263
- https://github.com/kubevirt/kubevirt/pull/8226
   diffs: 
   - tests/network/sriov.go: 
       - refactored test to use old functions prior to recent [refactor](https://github.com/kubevirt/kubevirt/pull/8299).
       - Add skip to test in case Multus is not updated with the updated network-status annotation.
       
This cherry pick PR fixes https://github.com/kubevirt/kubevirt/issues/7444 on `release-0.56` branch

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — APPROVED** [review]

**Review by EdDev — APPROVED** [review]

File: tests/network/sriov.go [hunk 1]
```diff
@@ -543,6 +545,11 @@ var _ = Describe("[Serial]SRIOV", func() {
 				vmi = startVmi(vmi)
 				vmi = waitVmi(vmi)
 
+				// Since there is no consistent way to tell if the network-status Multus annotation contains the
+				// PCI-Address data for SR-IOV interfaces in the version deployed, manually checking if it exists
+				if !networkStatusAnnotationContainsPCIData(vmi) {
```
Comment (AlonaKaplan) [inline tests/network/sriov.go:550]: Doesn't exist?

**Review by EdDev — APPROVED** [review]

**Comment by RamLavi** [comment]
[fixed](https://github.com/kubevirt/kubevirt/compare/910d7de11254be0d834f6fec98614c9c44aed2cf..cad60db172f92162d5ba8ff66ec2d2617774f410)
/hold cancel

**Pull request merged**